### ISSUE TITLE
Enabling Chunking

### ISSUE THREAD

**Issue by L-M-V** [BlueBrain/HighFive#76]
Hello,

Firstly I'd like to thank you for the HighFive library, it's been mostly a pleasure to work with. 

However I'm currently converting some older parallel HDF5 fortran code and have run into the following problem:

```
      call h5pcreate_f(H5P_DATASET_CREATE_F, chunkingPropertyID, result)
      call h5pset_chunk_f(chunkingPropertyID, numberOfDims, chunkDimensions, result)

      call h5dcreate_f(groupID, datasetName, H5T_IEEE_F64LE, fileID, datasetID, result, dcpl_id = chunkingPropertyID)
```

Normally it's possible to apply these properties after the creation by using the `getId()` function. However it looks like HDF5 only takes chunking as a parameter in the creation of the dataset (you can't set it later).

Is there another way to accomplish this?

**Comment by L-M-V** [comment]
Related to this:
Enabling Collective IO. At the moment I have to fall back to the C interface. 

```
   auto dataSetWritePropertyId = H5Pcreate(H5P_DATASET_XFER);

   // Enable collective I/O
   // Unfortunately no Highfive C++ wrapper for this yet, so we fall back to C
   h5pset_dxpl_mpio(dataSetWritePropertyId, H5FD_MPIO_COLLECTIVE);

   ...

   H5Dwrite(datasetId, H5T_NATIVE_DOUBLE, memoryDataspace.getId(), fileDataspace.getId(), dataSetWritePropertyId, output.memptr())
```

**Comment by jrs65** [comment]
I've patched HighFive to support chunked datasets and the ability to resize them (also the ability to pass in custom dataset creation properties). Hopefully I'll tidy that up tonight and create a pull request for it.

**Comment by L-M-V** [comment]
That sounds excellent @jrs65 . Thank you.

**Comment by ferdonline** [comment]
Hi. I think this is a great addition and the code looks clean to me.
@jrs65 Wold you like to create a pull request?

**Comment by ferdonline** [comment]
Implemented in #113
